project_euler/problem_135/sol1.py

```diff
@@ -0,0 +1,56 @@
+"""
+Project Euler Problem 135: https://projecteuler.net/problem=135
+
+Given the positive integers, x, y, and z,
+are consecutive terms of an arithmetic progression,
+the least value of the positive integer, n,
+for which the equation,
+x2 − y2 − z2 = n, has exactly two solutions is n = 27:
+
+342 − 272 − 202 = 122 − 92 − 62 = 27
+
+It turns out that n = 1155 is the least value
+which has exactly ten solutions.
+
+How many values of n less than one million
+have exactly ten distinct solutions?
+
+
+Taking x,y,z of the form a+d,a,a-d respectively,
+the given equation reduces to a*(4d-a)=n.
+Calculating no of solutions for every n till 1 million by fixing a
+,and n must be multiple of a.
+Total no of steps=n*(1/1+1/2+1/3+1/4..+1/n)
+,so roughly O(nlogn) time complexity.
+
+"""
+
+
+def solution(limit: int = 1000000) -> int:
+    """
+    returns the values of n less than or equal to the limit
+    have exactly ten distinct solutions.
+    >>> solution(100)
+    0
+    >>> solution(10000)
+    45
+    >>> solution(50050)
+    292
+    """
+    limit = limit + 1
+    freq = [0] * limit
+    for a in range(1, limit):
+        for n in range(a, limit, a):
+            d = a + n / a
+            if d % 4:  # d must be divisble by 4
+                continue
+            else:
+                d /= 4
+                if a > d and a < 4 * d:  # since x,y,z are positive integers
+                    freq[n] += 1  # so z>0 and a>d ,also 4d<a
+    count = 0
+    for n in range(1, limit):
+        if freq[n] == 10:
+            count += 1
+
+    return count
```

Comment:
Please add the `if __name__ == "__main__":` block as mentioned in project_euler/README.md

Reply:
And also adding  print(f"{solution() = }") ,the equal sign at the end is causing my pull request to fail at black tests, I am new pre-commit hooks and open source development. Need help.

Reply:
Yes